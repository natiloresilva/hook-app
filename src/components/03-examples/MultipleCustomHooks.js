import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import { useFetch } from '../../hooks/useFetch'
import '../02-useEffect/effects.css'

export const MultipleCustomHooks = () => {

    const {counter, increment, reset} = useCounter(1)
    counter === 31 && reset()

    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`)
  
    const {author, quote} = !!data && data[0]
    // La data en un primer momento viene null, entonces ===> !! data (!!null) nos retorna FALSE, por lo tanto no va a continuar. 
    //En ese caso author y quotes tendr√°n el valor de undefined
   
    return (
        <>
            <h1>Breaking Bad Quotes {counter}</h1>
            <hr/>

            {
                loading 
            ?
                (
                 <div className='alert alert-info text-center'>
                    Loading...
                </div>
                )
            :
                (
                <blockquote className='blockquote text-end'>
                    <p> {quote} </p>
                    <footer className='blockquote-footer'> {author} </footer>
                </blockquote>
                )
            }
           <button 
            className='btn btn-outline-info'
            onClick={() => increment()}
            >
             Next quote
            </button>

        </>
    )
}
