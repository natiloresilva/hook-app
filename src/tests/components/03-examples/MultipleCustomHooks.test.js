import { shallow } from 'enzyme'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
import { useCounter } from '../../../hooks/useCounter'
import { useFetch } from '../../../hooks/useFetch'
jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')


describe('Pruebas en <MultipleCustomHooks />', () => {

    beforeEach(() => {

        useCounter.mockReturnValue({
            counter: 10,
            increment: () => { }
        })

    })

    test('debe mostrarse correctamente el componente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar la informaciÃ³n ', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Walter White',
                quote: 'Say my name'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)
        //console.log(wrapper.html())

        expect(wrapper.find('.alert').exists()).toBeTruthy
        expect(wrapper.find('p').text().trim()).toBe('Say my name')
        expect(wrapper.find('footer').text().trim()).toBe('Walter White')

    })

})
