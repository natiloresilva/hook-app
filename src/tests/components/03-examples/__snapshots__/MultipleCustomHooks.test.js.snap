// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes 
    10
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-outline-info"
    onClick={[Function]}
  >
    Next quote
  </button>
</Fragment>
`;
